server/tables/tables.go

```diff
@@ -578,78 +578,73 @@ func PreAutoMigrate(db *gorm.DB) ([]PostAutoMigrateLogic, error) {
 		})
 	}
 
-	if m.HasTable("Invocations") && m.HasColumn(&Invocation{}, "invocation_uuid") {
-		// TODO(luluz): Disable migration code for now
-		// if db.Dialector.Name() == mysqlDialect {
-		// 		err := backfillInvocationUUIDIfNull(db)
-		//	return postMigrate, err
-		//}
-	}
-
 	// Populate invocation_uuid if the column doesn't exist.
-	if m.HasTable("Invocations") && !m.HasColumn(&Invocation{}, "invocation_uuid") {
+	if m.HasTable("Invocations") && m.HasColumn(&Invocation{}, "invocation_pk") {
 		if db.Dialector.Name() == sqliteDialect {
-			// TODO(luluz): Disable migration code for now
-			//postMigrate = append(postMigrate, func() error {
-			//	return postMigrateInvocationUUIDForSQLite(db)
-			//})
+			postMigrate = append(postMigrate, func() error {
+				return postMigrateInvocationUUIDForSQLite(db)
+			})
 		} else if db.Dialector.Name() == mysqlDialect {
-			// TODO(luluz): Disable migration code for now
-			// postMigrate = append(postMigrate, func() error {
-			// return postMigrateInvocationUUIDForMySQL(db)
-			// })
+			postMigrate = append(postMigrate, func() error {
+				return postMigrateInvocationUUIDForMySQL(db)
+			})
 		} else {
 			log.Warningf("Unsupported sql dialect: %q", db.Dialector.Name())
 		}
 	}
 	return postMigrate, nil
 }
 
-func backfillInvocationUUIDIfNull(db *gorm.DB) error {
-	if err := db.Exec(`
-		UPDATE Invocations SET invocation_uuid = UUID_TO_BIN(invocation_id) 
-		WHERE invocation_uuid IS NULL
-	`).Error; err != nil {
-		return err
+func updateInBatch(db *gorm.DB, baseQuery string, batchSize int64) error {
+	totalRows := int64(0)
+	for {
+		tx := db.Exec(baseQuery+" LIMIT ?", batchSize)
+		if err := tx.Error; err != nil {
+			return err
+		}
+		totalRows += tx.RowsAffected
+		log.Infof("updated %d rows in total for base query %q", totalRows, baseQuery)
+		if tx.RowsAffected == 0 {
+			return nil
+		}
 	}
-	return db.Exec(`
-		UPDATE TargetStatuses ts SET invocation_uuid=
-			(SELECT invocation_uuid FROM Invocations i 
-			WHERE i.invocation_pk=ts.invocation_pk)
-		WHERE invocation_uuid IS NULL
-	`).Error
 }
 
 func postMigrateInvocationUUIDForMySQL(db *gorm.DB) error {
-	if err := db.Exec(`UPDATE Invocations SET invocation_uuid = UUID_TO_BIN(invocation_id)`).Error; err != nil {
+	updateInvocationStmt := `UPDATE Invocations SET invocation_uuid = UNHEX(REPLACE(invocation_id, "-","")) WHERE invocation_uuid IS NULL`
+	err := updateInBatch(db, updateInvocationStmt, 10000)
```

Comment:
nit: inline the updateInBatch call into the if statement below

Reply:
Done